refactor(dns): unify symbol naming (#4505)
paritytech/rust-libp2p: transports/dns/CHANGELOG.md

## 0.40.1 - unreleased

> - Remove `Dns` prefix from types like `TokioDnsConfig` and `DnsConfig` in favor of modules that describe the different variants. Users are encouraged to import the `libp2p::dns` module and refer to types as `dns::tokio::Transport` and `dns::async_std::Transport`.

Reviewer: ```suggestion
- Remove `Dns` prefix from types like `TokioDnsConfig` and `DnsConfig` in favor of modules that describe the different variants.
  Users are encouraged to import the `libp2p::dns` module and refer to types as `dns::tokio::Transport` and `dns::async_std::Transport`.
```

> ## 0.41.0

Reviewer: ```suggestion
## 0.40.1 - unreleased
```

> - Rename types with `Dns` prefix (`TokioDnsConfig`, and `DnsConfig`)
> - Integrate `GenDnsConfig` into `Config`
>     See [PR 4505].

Reviewer: ```suggestion
- Remove `Dns` prefix from types like `TokioDnsConfig` and `DnsConfig` in favor of modules that describe the different variants. Users are encouraged to import the `libp2p::dns` module and refer to types as `dns::tokio::Transport` and `dns::async_std::Transport`.
  See [PR 4505].
```
